### ISSUE TITLE
Ipv6 zone indices in /etc/resolv.conf are incorrectly handled

### ISSUE THREAD

**Issue by terje2001** [docker/docker#28172]
**Description**
If the /etc/resolv.conf of the host machine contains a 'nameserver' field using IPv6 addressing including a zone index, the generated /etc/resolv.conf contains illegal information.

Host machine /etc/resolv.conf:
```
# This file is managed by systemd-resolved(8). Do not edit.
#
# This is a dynamic resolv.conf file for connecting local clients directly to
# all known DNS servers.
#
# Third party programs must not access this file directly, but only through the
# symlink at /etc/resolv.conf. To manage resolv.conf(5) in a different way,
# replace this symlink by a static file or a different symlink.
#
# See systemd-resolved.service(8) for details about the supported modes of
# operation for /etc/resolv.conf.

nameserver 192.168.2.1
nameserver fe80::c__d:c__f:__0:d36e%3
```
Note the "%3" in the last line.

Then this becomes, in the docker container:
```
# This file is managed by systemd-resolved(8). Do not edit.
#
# This is a dynamic resolv.conf file for connecting local clients directly to
# all known DNS servers.
#
# Third party programs must not access this file directly, but only through the
# symlink at /etc/resolv.conf. To manage resolv.conf(5) in a different way,
# replace this symlink by a static file or a different symlink.
#
# See systemd-resolved.service(8) for details about the supported modes of
# operation for /etc/resolv.conf.

nameserver 192.168.2.1
%3
```
(Note the illegal last line).

This causes, among other things, Rancher Server to fail badly during startup...

**Steps to reproduce the issue:**
1. Run latest Arch with ipv6
2. Start Centos or Ubuntu as a docker container
3. cat /etc/hosts to see illegal information

**Describe the results you received:**
Illegal content of "%3" as last line.

**Describe the results you expected:**
Simple: no "%3" as last line.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**
```
Client:
 Version:      1.12.1
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   23cf638
 Built:        Thu Aug 18 17:52:38 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.1
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   23cf638
 Built:        Thu Aug 18 17:52:38 2016
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
Containers: 3
 Running: 0
 Paused: 0
 Stopped: 3
Images: 83
Server Version: 1.12.1
Storage Driver: btrfs
 Build Version: Btrfs v3.17
 Library Version: 101
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: host bridge null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options: seccomp
Kernel Version: 4.8.6-2-ck
Operating System: Arch Linux
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 31.27 GiB
Name: jenna
ID: YNRS:UUF2:FEGR:FDCZ:UQXN:JOHV:DQVE:PEWE:GW6V:CVSC:ZCMK:K4ZT
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No cpu cfs quota support
WARNING: No cpu cfs period support
WARNING: No cpu shares support
Insecure Registries:
 127.0.0.0/8
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by justincormack** [comment]
ping @aboch

**Comment by justincormack** [comment]
Looks like this is the ipv6 matcher in libnetwork resolvconf.go...

```
        // This is not an IPv6 address verifier as it will accept a super-set of IPv6, and also
        // will *not match* IPv4-Embedded IPv6 Addresses (RFC6052), but that and other variants
        // -- e.g. other link-local types -- either won't work in containers or are unnecessary.
        // For readability and sufficiency for Docker purposes this seemed more reasonable than a
        // 1000+ character regexp with exact and complete IPv6 validation
        ipv6Address = `([0-9A-Fa-f]{0,4}:){2,7}([0-9A-Fa-f]{0,4})`
```

**Comment by thaJeztah** [comment]
Ping @sanimej PTAL

**Comment by sanimej** [comment]
I will fix the regex and may be we can take that fix for 1.13. Looking at the currently logic, it probably need some more changes. For an IPv6 enabled network we shouldn't copy the link-local resolvers from the host to the container.

**Comment by aboch** [comment]
Fixed by #29319
